cmd/gcp-controller-manager/node_annotator.go

```diff
@@ -286,6 +309,51 @@ func extractKubeLabels(instance *compute.Instance) (map[string]string, error) {
 	return parsedLabels, nil
 }
 
+func extractNodeTaints(instance *compute.Instance) ([]core.Taint, error) {
+	const kubeEnvKey = "kube-env"
+
+	if instance.Metadata == nil {
+		return nil, errNoMetadata
+	}
+
+	var kubeEnv *string
+	for _, item := range instance.Metadata.Items {
+		if item == nil || item.Key != kubeEnvKey {
+			continue
+		}
+		if item.Value == nil {
+			return nil, fmt.Errorf("instance %q had nil %q", instance.SelfLink, kubeEnvKey)
+		}
+		kubeEnv = item.Value
```

Comment:
Why doesn't this break after finding a valid item to assign? Is this some instance where we only want to assign the last valid item?

Reply:
Good catch. We can break it here. Added.